op_benchmarks/triton/bench_mha.py

```diff
@@ -0,0 +1,517 @@
+        seqlens_k = torch.full((Z, ), N_CTX_K)
+
+    # Calculate cumulative sequence lengths
+    cu_seqlens_q = torch.cat([torch.tensor([0], dtype=torch.int32), seqlens_q.cumsum(dim=0, dtype=torch.int32)])
+    cu_seqlens_k = torch.cat([torch.tensor([0], dtype=torch.int32), seqlens_k.cumsum(dim=0, dtype=torch.int32)])
+
+    cu_seqlens_q = cu_seqlens_q.to(device="cuda")
+    cu_seqlens_k = cu_seqlens_k.to(device="cuda")
+    # Initialize q, k, v with variable lengths
+    total_q = cu_seqlens_q[-1].item()
+    total_k = cu_seqlens_k[-1].item()
+
+    q = torch.randn((total_q, HQ, D_HEAD), dtype=dtype, device="cuda").normal_(mean=0., std=0.5).requires_grad_(requires_grad)
+    k = torch.randn((total_k, HK, D_HEAD), dtype=dtype, device="cuda").normal_(mean=0., std=0.5).requires_grad_(requires_grad)
+    v = torch.randn((total_k, HK, D_HEAD), dtype=dtype, device="cuda").normal_(mean=0., std=0.5).requires_grad_(requires_grad)
+    sm_scale = D_HEAD**-0.5
+    return q, k, v, cu_seqlens_q, cu_seqlens_k, sm_scale
+
+
+def mha_input_helper(Z, HQ, HK, N_CTX_Q, N_CTX_K, D_HEAD, dtype, requires_grad=True):
+    torch.manual_seed(20)
+
+    # Initialize q, k, v
+    # bshd layout supported
+    q_tensor_shape = (Z, N_CTX_Q, HQ, D_HEAD)
+    k_tensor_shape = (Z, N_CTX_K, HK, D_HEAD)
+
+    q = torch.randn(q_tensor_shape, dtype=dtype, device="cuda", requires_grad=requires_grad)
+    k = torch.randn(k_tensor_shape, dtype=dtype, device="cuda", requires_grad=requires_grad)
+    v = torch.randn(k_tensor_shape, dtype=dtype, device="cuda", requires_grad=requires_grad)
+
+    sm_scale = D_HEAD**-0.5
+    # max_seqlens_q = N_CTX_Q
+    # max_seqlens_k = N_CTX_K
+
+    return q, k, v, sm_scale
+
+
+def nonvarlen_benchmark_configs():
+    configs = [
+        (16, 16, 16, 1024, 1024),
+        (8, 16, 16, 2048, 2048),
+        (4, 16, 16, 4096, 4096),
+        (2, 16, 16, 8192, 8192),
+        (1, 16, 16, 16384, 16384),
+        (2, 48, 48, 1024, 1024),
+        (2, 48, 48, 2048, 1024),
+        (2, 48, 48, 4096, 8192),
+        (2, 48, 48, 8192, 4096),
+        (2, 48, 48, 16384, 8192),
+        (8, 16, 16, 1989, 15344),
+        (4, 16, 16, 4097, 163),
+        (2, 16, 16, 8122, 2159),
+        (1, 16, 16, 16281, 7),
+        (2, 48, 48, 1021, 1020),
+        (2, 48, 48, 2001, 2048),
+        (2, 48, 48, 3996, 9639),
+        (2, 48, 48, 8181, 1021),
+    ]
+    return configs
+
+
+def varlen_benchmark_configs():
+    configs = [
+        (2, 16, 4, 1024, 1024),
+        (8, 16, 2, 2048, 2048),
+        (4, 16, 8, 4096, 4096),
+        (2, 16, 4, 8192, 8192),
+        (2, 16, 8, 16384, 16384),
+        (2, 48, 12, 1024, 1024),
+        (2, 48, 24, 2048, 2048),
+        (2, 48, 8, 4096, 4096),
+        (2, 48, 4, 8192, 8192),
+        (2, 48, 2, 16384, 16384),
+        (2, 64, 32, 1024, 1024),
+        (4, 64, 16, 2048, 2048),
+        (4, 64, 8, 4096, 4096),
+        (4, 64, 32, 8192, 8192),
+        (4, 128, 16, 16384, 16384),
+    ]
+    return configs
```

Comment:
I wonder if we need this or only the model shapes are enough? @rahulbatra85 could you confirm?

Reply:
@Chi-Chu319  For now, this good. We can add/modify as we need.